codalab/apps/web/utils.py

```diff
@@ -238,12 +237,26 @@ def get_filefield_size(obj, attr, aws_attr=None, s3direct=False):
                 logger.error(e)
                 # If we hit an exception this way, and we're using S3, try the other method.
                 if settings.USE_AWS:
-                    obj_summary = s3.ObjectSummary(attr_obj.storage.bucket.name, attr_obj.name)
-                    if obj_summary:
-                        size = obj_summary.size
+                    size = get_size_from_summary(attr_obj.storage.bucket.name, attr_obj.name)
     # Always make sure we return at least 0
     return size or 0
 
+def get_size_from_summary(bucket_name, key):
+    size = None
+    # Test if file exists, since it's an observed bug that the object exists but no file is present in the storage
+    try:
+        obj_summary = s3.ObjectSummary(bucket_name, key)
+        if obj_summary:
+            size = obj_summary.size
+    except ClientError as e:
+        if e.response['Error']['Code'] == "404":
+            # The object does not exist.
+            logger.error("File was not found in storage. File: {0}; Storage: {1}".format(bucket_name, key))
+        else:
+            # Something else has gone wrong.
+            raise
```

Comment:
I guess we are interested in catching 404 error (when the object does not exist). But is this `raise` syntax correct? Will it provide the complete back traces if called? Also, isn't the whole point of this function to return None if a bug is encountered?

Maybe we should just remove lines 255 - 257

Reply:
Indeed we are trying to catch a Client error which has an error code 404. But you are right that in case we catch a Client error with another error code, the trace back won't be informative. I'll apply some change to the code.

Reply:
Yes I get it but I think we should return None if an error is encountered, hence removing lines 255 - 257.

If we raise an error here, this means that the views "My submissions" and "All submissions" will bug.

Reply:
I approve your change since you still get the full traceback, thus debugging other cases remain possible.

Reply:
Shouldn't you also replace `raise` by `logger.error(traceback.format_exc())` ?